plutus-pab-executables/demo/pab-nami/client/default.nix

```diff
@@ -5,35 +5,64 @@ let
 
   pab-setup-invoker = haskell.packages.plutus-pab-executables.components.exes.plutus-pab-setup;
 
-  # TODO: Use the PS generator in the demo app
-  generated-purescript = pkgs.runCommand "pab-nami-demo-purescript" { } ''
-    mkdir $out
-    ${pab-nami-demo-generator}/bin/plutus-pab-nami-demo-generator --output-dir $out
-    cp ${builtins.path { name = "tidyrc.json"; path = ../../../../.tidyrc.json; } } $out/.tidyrc.json
-    cp ${builtins.path { name = "tidyoperators"; path = ../../../../.tidyoperators; } } $out/.tidyoperators
-    cd $out
-    ${purs-tidy}/bin/purs-tidy format-in-place $out
-    rm $out/.tidyrc.json
-    rm $out/.tidyoperators
-  '';
-
   generate-purescript = pkgs.writeShellScriptBin "pab-nami-demo-generate-purs" ''
-    generatedDir=./generated
+    if [ ! -d plutus-pab-executables ]; then 
+      echo Please run pab-nami-demo-generate-purs from the root of the repository
+      exit 1
+    fi
+
+    generatedDir=./plutus-pab-executables/demo/pab-nami/client/generated
     rm -rf $generatedDir
-    $(nix-build ../../../../default.nix -A pab-nami-demo.pab-nami-demo-generator)/bin/plutus-pab-nami-demo-generator --output-dir $generatedDir
-    cd ../../../..
-    echo Formatting files...
-    ${purs-tidy}/bin/purs-tidy format-in-place ./plutus-pab-executables/demo/pab-nami/client/generated
-    echo Done: formatted
+
+    echo Generating purescript files in $generatedDir
+    ${pab-nami-demo-generator}/bin/plutus-pab-nami-demo-generator --output-dir $generatedDir
+    echo -e Done generating purescript files
+
+    echo Formatting purescript files in $generatedDir
+    ${purs-tidy}/bin/purs-tidy format-in-place $generatedDir
+    echo Done formatting purescript files
   '';
 
   start-backend = pkgs.writeShellScriptBin "pab-nami-demo-server" ''
+    if [ ! -d plutus-pab-executables ]; then 
```

Comment:
Here the user is notified about the age of the generated files, and reminded that they can run -g to regenerete them.
If this is the first time they run it, and there is no generated folder, then it is created first.